### ISSUE TITLE
Ongoing problems with environment

### ISSUE THREAD

**Issue by Tomkourou** [pypsa-meets-africa/pypsa-africa-hackathon#8]
Re my last issue I have now managed to install the conda environment both in windows and in a linux virtual machine using ubuntu. 

Nevertheless there are still issues in both that remain. 

1. The installation took way too long, about 5 hours for the environment to solve. 
2. When I run the snakemake -j1 retrieve_databundle_light I get the following error: 
```
This is the repository path:  /home/thomas/Personal_projects/pypsa-africa
Had to go 0 folder(s) up.
INFO:__main__:Retrieving data from GoogleDrive.
Downloading 1he31BBLtdemZt2dmBOwUCbP_jVuI3KS8 into ./resources.zip... Done.
Unzipping.../home/thomas/anaconda3/envs/pypsa-africa/lib/python3.9/site-packages/google_drive_downloader/google_drive_downloader.py:78: UserWarning: Ignoring `unzip` since "1he31BBLtdemZt2dmBOwUCbP_jVuI3KS8" does not look like a valid zip file
  warnings.warn('Ignoring `unzip` since "{}" does not look like a valid zip file'.format(file_id))
INFO:__main__:Download data to './resources' from cloud 'https://drive.google.com/file/d/1he31BBLtdemZt2dmBOwUCbP_jVuI3KS8/view?usp=sharing'.
Downloading 1jyFPqcBYbt8_dQvqpw1CAoU-lN0uZOKK into ./data.zip... Done.
Unzipping.../home/thomas/anaconda3/envs/pypsa-africa/lib/python3.9/site-packages/google_drive_downloader/google_drive_downloader.py:78: UserWarning: Ignoring `unzip` since "1jyFPqcBYbt8_dQvqpw1CAoU-lN0uZOKK" does not look like a valid zip file
  warnings.warn('Ignoring `unzip` since "{}" does not look like a valid zip file'.format(file_id))
INFO:__main__:Download data to './data' from cloud 'https://drive.google.com/file/d/1jyFPqcBYbt8_dQvqpw1CAoU-lN0uZOKK/view?usp=sharing'.
Downloading 1-Njs7BqG0YE5QwBHj0zgkdicb5IQvQCh into ./cutouts.zip... Done.
Unzipping...Done.
INFO:__main__:Download data to './cutouts' from cloud 'https://drive.google.com/file/d/1-Njs7BqG0YE5QwBHj0zgkdicb5IQvQCh/view?usp=sharing'.
Waiting at most 5 seconds for missing files.
MissingOutputException in line 50 of /home/thomas/Personal_projects/pypsa-africa/Snakefile:
Job Missing files after 5 seconds:
resources/ssp2-2.6/2030/era5_2013/Africa.nc
data/raw/copernicus/PROBAV_LC100_global_v3.0.1_2019-nrt_Discrete-Classification-map_EPSG-4326.tif
data/raw/gebco/GEBCO_2021_TID.nc
data/raw/eez/eez_v11.gpkg
data/raw/hydrobasins/hybas_lake_af_lev04_v1c.shp
data/costs.csv
data/raw/landcover
This might be due to filesystem latency. If that is the case, consider to increase the wait time with --latency-wait.
Job id: 0 completed successfully, but some output files are missing. 0
Removing output files of failed job retrieve_databundle_light since they might be corrupted:
cutouts/africa-2013-era5-tutorial.nc
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
The code used to generate one or several output files has changed:
    To inspect which output files have changes, run 'snakemake --list-code-changes'.
    To trigger a re-run, use 'snakemake -R $(snakemake --list-code-changes)'.
The input used to generate one or several output files has changed:
    To inspect which output files have changes, run 'snakemake --list-input-changes'.
    To trigger a re-run, use 'snakemake -R $(snakemake --list-input-changes)'.
The params used to generate one or several output files has changed:
    To inspect which output files have changes, run 'snakemake --list-params-changes'.
    To trigger a re-run, use 'snakemake -R $(snakemake --list-params-changes)'.
```
I have tried to add latency with --latency-wait to no avail. Testing the retrieve_databundle_light.py it seems to be a problem with gdd or perhaps with the google drive folder itself. My understanding is that it's not managing the download of the zip file. 
3. Graphviz is required to use dot in snakemake --dag solve_all_networks | dot -Tpng -o workflow.png however when I try to install graphviz into my environment I get the following error: 
````
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - conda-forge/noarch::geoviews==1.9.4=pyhd8ed1ab_0
  - conda-forge/noarch::datashader==0.13.0=pyh6c4a22f_0
  - conda-forge/noarch::pypsa==0.19.1=pyhd8ed1ab_0
  - conda-forge/noarch::rioxarray==0.10.1=pyhd8ed1ab_0
  - conda-forge/noarch::atlite==0.2.5=pyhd8ed1ab_0
\ failed with initial frozen solve. Retrying with flexible solve.
````
Look forward to your thoughts!

**Comment by pz-max** [comment]
Hi @Tomkourou, Thanks for that issue :)
1. Regarding the conda environment. It should actually take max. 10-15min. I guess you could update anaconda, clean your base environment and setup again the environment to accelerate the installation. You can find very useful basic conda knowledge [here](https://towardsdatascience.com/conda-essential-concepts-and-tricks-e478ed53b5b)  

2. The `retrieve_databundle_light.py` script got recently an update. I can recommend installing pypsa-africa from scratch at the original repository https://github.com/pypsa-meets-africa/pypsa-africa. (Also using the new environment given at the original repository). If your problem persists, I can recommend installing a Linux distribution called Ubuntu (without removing windows). Since two or three years this is now super simple. Every Windows PC offers Ubuntu at the `software store` that you can probably access over your home menu. > Search than for ubuntu and install it. This should give you a Linux terminal, where you can try the pypsa-africa installation again.

3. Conda always works smoothly when you install the environment for the first time (see again [conda basics](https://towardsdatascience.com/conda-essential-concepts-and-tricks-e478ed53b5b)). Every further installation might end up in dependency issues as you are facing. You can try two things: i) remove pypsa-africa environment, add to the pypsa-africa\envs\environment.yaml the name `graphviz ` and install than as usual the pypsa-africa environment ii) an alternative is to install graphviz with pip `pip install graphviz`. 

Please update us if you solved the issue

**Comment by Tomkourou** [comment]
Hi @pz-max, I have actually got a completely clean install in my base environment in both my unbuntu and windows environments. I will try installing from the pypsa-africa repo directly and let you know how I get on! Thanks :D

**Comment by Tomkourou** [comment]
@pz-max just a quick update: 

1. Not sure what the issue is here but as I previously mentioned I tested it both in ubuntu and windows with completely clean base environments. 
2. I used the original pypsa-africa repo and ran snakemake -j1 retrieve_databundle_light and this worked perfectly. I did this using the environment I installed from the hackathon. For more information I noticed that the gdd downloader was downloading empty zip files and this is what was breaking the workflow. Not sure where this problem came from but I guess if you push the update to the hackathon repo it will be fixed. 
3. pip install graphviz worked perfectly. 

Thanks for your help!
